src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useTestResultsAggregates/useTestResultsAggregates.tsx

```diff
@@ -0,0 +1,140 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import { MeasurementInterval } from 'pages/RepoPage/shared/constants'
+import { RepoNotFoundErrorSchema } from 'services/repo'
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+const TestResultsAggregatesSchema = z.object({
+  owner: z
+    .object({
+      plan: z
+        .object({
+          value: z.string(),
+        })
+        .nullable(),
+      repository: z.discriminatedUnion('__typename', [
+        z.object({
+          __typename: z.literal('Repository'),
+          private: z.boolean().nullable(),
+          defaultBranch: z.string().nullable(),
+          testAnalytics: z
+            .object({
+              testResultsAggregates: z.object({
+                totalDuration: z.number(),
```

Comment:
In too deep with this now but would be nice to have units on these `duration` fields next time. Maybe with the polish epic can add comments indicating these (`totalDuration`, `slowestTestsDuration`) represent seconds either here and/or in `api`

Reply:
For sure, could be a nice-to-have in the GQ schema playground too! [Add comments indicating these (totalDuration, slowestTestsDuration) represent seconds #2879](https://github.com/codecov/engineering-team/issues/2879)